modelon/impact/client/experiment_definition.py

```diff
@@ -519,13 +488,18 @@ def __init__(
         self._initialize_from_experiment = None
         self._initialize_from_case = None
         self._initialize_from_external_result = None
+        self._expansion = FullFactorial()
 
     def validate(self):
         raise NotImplementedError(
             "Validation is not supported for SimpleModelicaExperimentDefinition class"
         )
 
-    def with_modifiers(self, modifiers: dict = None):
+    def with_modifiers(
+        self,
+        modifiers: Optional[Dict[str, Any]] = None,
+        expansion: Optional[ExpansionAlgorithm] = FullFactorial(),
```

Comment:
This is very dangerous because you can for example override previous choices with this default value. We really should have a separate method for updating the expansion algorithm. Like:
```
exp_def.with_modifiers({'inertia1.J': Beta(1, 2)}).with_expansion(Sobol(5))
```